src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -849,49 +849,6 @@ ep_rt_mono_init_finish (void)
 	mono_error_cleanup (error);
 }
 
-void
-ep_rt_mono_fini (void)
-{
-	// Avoid cleaning up resources to prevent cleaning up out from under running
-	// threads.
-	if (!mono_runtime_is_shutting_down ())
-		return;
-
-	ep_rt_mono_runtime_provider_fini ();
-	ep_rt_mono_profiler_provider_fini ();
-
-	if (_eventpipe_initialized)
-		mono_rand_close (_rand_provider);
-
-	_rand_provider = NULL;
-	_eventpipe_initialized = FALSE;
-
-	_ep_rt_mono_runtime_initialized = FALSE;
-
-	if (_ep_rt_mono_default_profiler_provider) {
-		mono_profiler_set_runtime_initialized_callback (_ep_rt_mono_default_profiler_provider, NULL);
-		mono_profiler_set_thread_started_callback (_ep_rt_mono_default_profiler_provider, NULL);
-		mono_profiler_set_thread_stopped_callback (_ep_rt_mono_default_profiler_provider, NULL);
-	}
-	_ep_rt_mono_default_profiler_provider = NULL;
```

Comment:
I think we should still unregister the profiling callbacks.  It will  prevent other potential conflicts between shutdown and EP